File: SenRestApi/Models/Produit.cs
using System.Xml.Serialization;

namespace SenRestApi.Models;

[XmlType(Namespace = "http://service.soap.exemple.com/")]
public class Produit
{
    [XmlElement]
    public long Id { get; set; }

    [XmlElement]
    public string? Nom { get; set; }

    [XmlElement]
    public double Prix { get; set; }

    [XmlElement]
    public int Quantite { get; set; }

    [XmlElement]
    public long TypeId { get; set; }
}

File: soap-server-java/src/main/java/com/exemple/soap/model/Produit.java
package com.exemple.soap.model;

import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class Produit {
    private Long id;
    private String nom;
    private Double prix;
    private Integer quantite;
    private Long typeId;

    public Produit() {}

    public Produit(Long id, String nom, Double prix, Integer quantite, Long typeId) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.quantite = quantite;
        this.typeId = typeId;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }
    public Double getPrix() { return prix; }
    public void setPrix(Double prix) { this.prix = prix; }
    public Integer getQuantite() { return quantite; }
    public void setQuantite(Integer quantite) { this.quantite = quantite; }
    public Long getTypeId() { return typeId; }
    public void setTypeId(Long typeId) { this.typeId = typeId; }

    @Override
    public String toString() {
        return "Produit{id=" + id + ", nom='" + nom + "', prix=" + prix + ", quantite=" + quantite + ", typeId=" + typeId + "}";
    }
}
